spring-cloud-dataflow-docs/src/main/asciidoc/architecture.adoc

```diff
@@ -130,6 +130,12 @@ Taps can be used to ‘listen in’ to the data that if flowing across any of th
 === Concurrency
 For an application that will consume events, Spring Cloud stream exposes a concurrency setting that controls the size of a thread pool used for dispatching incoming messages.  See the link:http://docs.spring.io/spring-cloud-stream/docs/1.0.2.RELEASE/reference/htmlsingle/index.html#_consumer_properties[Consumer properties] documentation for more information.
 
+[[arch-streams-partitioning]]
+=== Partitioning
+In some situations (for example to benefit from scaling out), applications can be deployed using multiple instances and
+benefit from data partitioning (whereby some subset of data always goes to the same instance according to some
+business rule). Please refer to <<passing_stream_partition_properties>> to learn more.
+
```

Comment:
I believe we should simply clarify that dynamic scaling is not supported with partitioning. This goes beyond a simple 'binder limitation'. Scaling up with semantic partitioning needs to be part of the more general stateful processing story. It's not only the binder that has to support it, but the state store as well (which needs to be part of the SCSt application model). When we scale up/down we effectively shuffle data across instances so we need to ensure that consumers are stopped, and the state is consistently flushed before all consumers begin consuming again. Also, in order for the process to be effective, the state store needs to be partitioned as well, and we need to ensure that the partitioning strategy of the state store after scaling matches the post-scaling data allocation scheme of the consumers. At this time, it is significantly simpler to rely on stopping and restarting the streams to ensure consistency in what concerns application lifecycle and state flushing. If we are to dynamically scaling with partitioning, we need to do this in conjunction with first-class stateful processing support.